core/unit_test/TestReduce.hpp

```diff
@@ -86,12 +88,27 @@ class ReduceFunctor {
     dst.value[2] += src.value[2];
   }
 
+  KOKKOS_INLINE_FUNCTION
+  void join(const ReducerTag, volatile value_type& dst,
+            const volatile value_type& src) const {
```

Comment:
Probably should either do something different from the overload that does not take a work tag or define another class.

Reply:
I opted for not deriving from `ReduceFunctor`.